### PR TITLE
Remove redundant jitserver build platforms

### PR DESCRIPTION
**Pull request by keithc-ca**
* add '--enable-jitserver' for s390x_linux
* remove *_linux_jit platforms whose bases specify '--enable-jitserver'
* remove *linuxjit short names

### CODE REVIEW

File: buildenv/jenkins/variables/defaults.yml [hunk 1]
```diff
@@ -232,17 +227,14 @@ s390x_linux:
     11: 'linux-s390x-normal-server-release'
   node_labels:
     build: 'ci.role.build && hw.arch.s390x && sw.os.rhel.7'
+  extra_configure_options:
+    all: '--enable-jitserver'
```
Comment (pshipton) [inline buildenv/jenkins/variables/defaults.yml:231]: This should be unnecessary once https://github.com/ibmruntimes/openj9-openjdk-jdk11/pull/540 is approved and rolled out for all versions.

Comment (keithc-ca) [reply buildenv/jenkins/variables/defaults.yml:231]: Yes, and it can be removed from x and p linux platforms then.

**Comment by keithc-ca** [comment]
> We don't want to remove the jitserver specs, it would break jitserver testing Note the `test_flags: 'JITAAS'` in the `jitserver` definition.

Ok, I reworked the changes so the specs extend `jitserver` to retain that test flag.

**Comment by keithc-ca** [comment]
> I expect it means that all testing will be done using jitserver, and not with the local JIT. This is undesirable.

Agreed; closing.

**Pull request closed without merging**